src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -36,86 +37,76 @@ public override void Initialize(AnalysisContext context)
         {
-            ArgumentAnalyzer.ReportProblems(context, availableFunctions, calledFunctions);
             NameAnalyzer.ReportProblems(context, availableFunctions, calledFunctions);
-            FunctionReturnTypeAnalyzer.ReportProblems(context, availableFunctions, calledFunctions);
+            ArgumentAnalyzer.ReportProblems(context, semanticModel, availableFunctions, calledFunctions);
+            FunctionReturnTypeAnalyzer.ReportProblems(context, semanticModel, availableFunctions, calledFunctions);
         }
 
         public void FindActivityCall(SyntaxNodeAnalysisContext context)
         {
-            if (context.Node is InvocationExpressionSyntax invocationExpression && 
-                SyntaxNodeUtils.IsInsideFunction(invocationExpression) && 
-                IsCallActivityInvocation(invocationExpression))
+            if (context.Node is InvocationExpressionSyntax invocationExpression
+                && SyntaxNodeUtils.IsInsideFunction(invocationExpression)
+                && IsActivityInvocation(invocationExpression))
             {
-                if (!TryGetFunctionNameFromCallActivityInvocation(invocationExpression, out SyntaxNode functionNameNode))
+                StoreSemanticModel(context);
+
+                if (!TryGetFunctionNameFromActivityInvocation(invocationExpression, out SyntaxNode functionNameNode))
                 {
                     //Do not store ActivityFunctionCall if there is no function name
                     return;
                 }
 
-                var returnTypeName = GetReturnTypeNameFromCallActivityInvocation(context, invocationExpression);
+                SyntaxNodeUtils.TryGetTypeArgumentNode((MemberAccessExpressionSyntax)invocationExpression.Expression, out SyntaxNode returnTypeNode);
 
-                string inputTypeName = null;
-                if (TryGetInputNodeFromCallActivityInvocation(invocationExpression, out SyntaxNode inputNode))
-                {
-                    var inputType = context.SemanticModel.GetTypeInfo(inputNode).Type;
-                    inputTypeName = GetQualifiedTypeName(inputType);
-                }
+                TryGetInputNodeFromCallActivityInvocation(invocationExpression, out SyntaxNode inputNode);
 
                 calledFunctions.Add(new ActivityFunctionCall
                 {
                     Name = functionNameNode.ToString().Trim('"'),
                     NameNode = functionNameNode,
                     ParameterNode = inputNode,
-                    ParameterType = inputTypeName,
-                    ExpectedReturnType = returnTypeName,
+                    ReturnTypeNode = returnTypeNode,
                     InvocationExpression = invocationExpression
                 });
             }
         }
 
-        private bool IsCallActivityInvocation(InvocationExpressionSyntax invocationExpression)
+        private void StoreSemanticModel(SyntaxNodeAnalysisContext context)
```

Comment:
nit: consider `SaveSemanticModel` or `SetSemanticModel` as the method name. `Store` usually implies some kind of persistent storage.